code/modules/client/version_management.dm

```diff
@@ -0,0 +1,65 @@
+/**
+  * # BYOND Version
+  *
+  * Datum to easier represent a BYOND version in code
+  *
+  * Used for the version warning/blacklist system
+  * Provides an easy means of tacking information onto a build, and a flag whether to just warn the client, or blacklist it entirely
+  *
+  */
+/datum/byond_version
+	/// Response type (Either CLIENT_BUILD_WARNING or CLIENT_BUILD_BLACKLISTED)
+	var/response_type
+	/// Details on why this build is warned/blacklisted
+	var/details
+
+/**
+  * Proc to format BYOND version info and datumise it
+  *
+  * Basically simplifies creation of BYOND build info, and returns a datum with all the info
+  *
+  * Arguments:
+  * * restype - What response to give the client if they connect with this build (Either CLIENT_BUILD_WARNING or CLIENT_BUILD_BLACKLISTED)
+  * * info - Details on why this version is flagged
+  */
+/proc/format_byond_version_info(restype, info)
+	var/datum/byond_version/BV = new
+	BV.response_type = restype
+	BV.details = info
+	return BV
+
+/**
+  * Proc to register all flagged BYOND versions
+  *
+  * This proc is called from world/New(), and is for registering BYOND versions
+  * Add versions to flag into this proc, following the format of the others
+  * Please keep these in build version order. Please.
+  * VERSION NUMBER MUST BE A STRING BECAUSE OTHERWISE IT USES INDEXES NOT ASSOCIATIONS
+  */
+/proc/register_flagged_byond_versions()
+	// Do not put a . at the end of the details, otherwise it will double-period when the client sees it
+	GLOB.flagged_byond_versions["1407"] = format_byond_version_info(CLIENT_BUILD_BLACKLISTED, "This client version contains a bug preventing display overrides from working, which leads to clients being able to see atoms they shouldn't be able to see")
```

Comment:
Shouldn't this be set in config? Otherwise we have to do PRs whenever we need to add a new blacklisted version. Also, these reasons need to be understandable by normal players in addition to having enough info for admins/etc to realize how bad the problem is.